src/features/contacts/contacts-table.tsx

```diff
@@ -1,60 +1,185 @@
-import { Button } from "@mui/material";
-import { DataGrid, GridColDef } from "@mui/x-data-grid";
+import {
+  DataGrid,
+  getGridStringOperators,
+  GridColDef,
+  GridFilterModel,
+  GridSortModel,
+} from "@mui/x-data-grid";
+import EditIcon from "@mui/icons-material/Edit";
+import ArrowForwardIcon from "@mui/icons-material/ArrowForward";
 import { ContactDetailsDto } from "lib/network/swagger-client";
-import { ContactsTableContainer } from "./index.styled";
+import {
+  ActionButtonContainer,
+  ContactsTableContainer,
+  EditIconContainer,
+  ForwardIconContainer,
+} from "./index.styled";
+import { CoreModule, getEditModuleRoute } from "lib/router";
+import { useMemo, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { Avatar, ListItem, ListItemAvatar, ListItemText } from "@mui/material";
 
 type ContactsTableProps = {
   contacts?: ContactDetailsDto[];
+  pageSize?: number;
+  totalRowCount: number;
+  setSortColumn: (sortColumn: string) => void;
+  setSortOrder: (sortOrder: string) => void;
+  setPageSize: (pageSize: number) => void;
+  setSkipLimit: (skipLimit: number) => void;
+  setFilterField: (filterField: string) => void;
+  setFilterFieldValue: (fieldValue: string) => void;
 };
 
-const empty = [] as const;
-
-const columns: GridColDef<ContactDetailsDto>[] = [
-  {
-    field: "name",
-    headerName: "Name",
-    minWidth: 150,
-    flex: 1,
-    renderCell: ({ row }) => (
-      <div>
-        <div>{`${row.firstName} ${row.lastName}`}</div>
-        <div>{row.email}</div>
-      </div>
-    ),
-  },
-  {
-    field: "location",
-    headerName: "Location",
-  },
-  {
-    field: "orders",
-    headerName: "Orders",
-  },
-  {
-    field: "spent",
-    headerName: "Spent",
-  },
-  {
-    field: "actions",
-    headerName: "Actions",
-    align: "right",
-    headerAlign: "right",
-    sortable: false,
-    disableColumnMenu: true,
-    renderCell: () => {
-      return (
-        <div>
-          <Button>edit</Button>
-        </div>
-      );
+export const ContactsTable = ({
+  contacts,
+  setPageSize,
+  pageSize,
+  setSkipLimit,
+  totalRowCount,
+  setSortColumn,
+  setSortOrder,
+  setFilterField,
+  setFilterFieldValue,
+}: ContactsTableProps) => {
+  const [page, setPage] = useState(0);
+
+  const navigate = useNavigate();
+
+  const empty = [] as const;
+
+  const handleEditClick = (id: number) => {
+    navigate(getEditModuleRoute(id));
+  };
+
+  const handleForwardClick = () => {
+    //TODO
```

Comment:
don't left empty ToDo, please fiil in what you plan to do here

Reply:
Updated.